server/index.js

```diff
@@ -48,14 +48,32 @@ async function setupApp() {
 
   });
 
+  app.post('/users', async (req, res) => {
+    try {
+      const { name, email, password, birthdate, notification_preference, biography, terms_agreed, newsletter_subscribed } = req.body;
+      const user = await pgClient.query(`SELECT * FROM  messages WHERE email = $1`, [email]);
+      if(user.length !== 0){
+        return res.json({ error: 'email already exists' });
+      }
+      if (user.password.length < 6){
+        return res.status(401).json({ message: 'password has to be more than 6 characters' });
+      }
+
+      const userData = await pgClient.query(`INSERT INTO users (name, email, birthdate, notification_preference, terms_agreed, newsletter_subscribed) VALUES ($1, $2, $3, $4, $5, $6) RETURNING *`, [name, email, password, birthdate, notification_preference, biography, terms_agreed, newsletter_subscribed ]);
```

Comment:
The frontend names are different (you have to check name attributes in signup.js). 🤷‍♂️ Just gotta use correct names.

```suggestion
      const userData = await pgClient.query(`INSERT INTO users (name, email, birthdate, notification_preference, terms_agreed, newsletter_subscribed) VALUES ($1, $2, $3, $4, $5, $6) RETURNING *`, [name, email, password, birthdate, notificationPreference, biography, agreeToTerms, subscribeToNewsLetter ]);
```